src/Parlot/Fluent/NumberLiteral.Net80.cs

```diff
@@ -0,0 +1,138 @@
+﻿#if NET8_0_OR_GREATER
+using Parlot.Compilation;
+using System;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Numerics;
+using System.Reflection;
+
+namespace Parlot.Fluent
+{
+    public static class NumberLiteral
+    {
+        public const char DefaultDecimalSeparator = '.';
+        public const char DefaultGroupSeparator = ',';
+
+        public static NumberLiteral<T> CreateNumberLiteralParser<T>(NumberOptions numberOptions = NumberOptions.Number, char decimalSeparator = DefaultDecimalSeparator, char groupSeparator = DefaultGroupSeparator)
+            where T : INumber<T>
+        {
+            return new NumberLiteral<T>(numberOptions, decimalSeparator, groupSeparator);
+        }
+    }
+
+    public sealed class NumberLiteral<T> : Parser<T>, ICompilable
+        where T : INumber<T>
+    {
+        private const char DefaultDecimalSeparator = '.';
+        private const char DefaultGroupSeparator = ',';
+
+        private readonly NumberOptions _numberOptions;
+        private readonly char _decimalSeparator;
+        private readonly char _groupSeparator;
+        private readonly NumberStyles _numberStyles;
+        private readonly CultureInfo _culture = CultureInfo.InvariantCulture;
+
+        private static readonly MethodInfo _tryParseMethodInfo = typeof(T).GetMethod(nameof(INumber<T>.TryParse), [typeof(ReadOnlySpan<char>), typeof(NumberStyles), typeof(IFormatProvider), typeof(T).MakeByRefType()]);
+
+        public NumberLiteral(NumberOptions numberOptions = NumberOptions.Number, char decimalSeparator = DefaultDecimalSeparator, char groupSeparator = DefaultGroupSeparator)
+        {
+            _numberOptions = numberOptions;
+            _decimalSeparator = decimalSeparator;
+            _groupSeparator = groupSeparator;
+            _numberStyles = _numberOptions.ToNumberStyles();
+            
+            if (decimalSeparator != CultureInfo.InvariantCulture.NumberFormat.NumberDecimalSeparator[0] || 
```

Comment:
Maybe could check `decimalSeparator != DefaultDecimalSeparator || groupSeparator != DefaultGroupSeparator )`, defaults seem to be the invariant versions